garage/envs/normalized_env.py

```diff
@@ -75,39 +81,30 @@ def __setstate__(self, d):
         self._obs_mean = d["_obs_mean"]
         self._obs_var = d["_obs_var"]
 
-    @property
-    @overrides
-    def action_space(self):
-        if isinstance(self._wrapped_env.action_space, gym.spaces.Box):
-            ub = np.ones(self._wrapped_env.action_space.shape)
-            return gym.spaces.Box(-1 * ub, ub, dtype=np.float32)
-        return self._wrapped_env.action_space
-
     @overrides
     def step(self, action):
-        if isinstance(self._wrapped_env.action_space, gym.spaces.Box):
+        if isinstance(self.action_space, gym.spaces.Box):
             # rescale the action
-            lb, ub = self._wrapped_env.action_space.bounds
-            scaled_action = lb + (action + 1.) * 0.5 * (ub - lb)
-            scaled_action = np.clip(scaled_action, lb, ub)
+            lb, ub = self.action_space.low, self.action_space.high
+            if lb != -np.inf or ub != -np.inf:
+                scaled_action = lb + (action + 1.) * 0.5 * (ub - lb)
+                scaled_action = np.clip(scaled_action, lb, ub)
+            else:
+                scaled_action = action
         else:
             scaled_action = action
-        wrapped_step = self._wrapped_env.step(scaled_action)
-        next_obs, reward, done, info = wrapped_step
+
+        next_obs, reward, done, info = self.env.step(scaled_action)
+
         if self._normalize_obs:
             next_obs = self._apply_normalize_obs(next_obs)
         if self._normalize_reward:
             reward = self._apply_normalize_reward(reward)
-        return Step(next_obs, reward * self._scale_reward, done, **info)
 
-    def __str__(self):
-        return "Normalized: %s" % self._wrapped_env
+        return next_obs, reward * self._scale_reward, done, info
 
-    # def log_diagnostics(self, paths):
-    #     print "Obs mean:", self._obs_mean
-    #     print "Obs std:", np.sqrt(self._obs_var)
-    #     print "Reward mean:", self._reward_mean
-    #     print "Reward std:", np.sqrt(self._reward_var)
+    def __str__(self):
+        return "Normalized: %s" % self.env
```

Comment:
Doesn't gym already provide a nice version of this?

https://github.com/openai/gym/blob/master/gym/core.py#L285